packages/contracts/contracts/MarginlyPool.sol

```diff
@@ -785,35 +798,75 @@ contract MarginlyPool is IMarginlyPool {
+    FP96.FixedPoint memory onePlusFee = FP96.fromRatio(params.fee, WHOLE_ONE).div(secondsInYear).add(FP96.one());
+
+    // FEE(dt) = (1 + fee)^dt
+    FP96.FixedPoint memory feeDt = FP96.powTaylor(onePlusFee, secondsPassed);
+
+    uint256 discountedBaseFee;
+    uint256 discountedQuoteFee;
+
     if (discountedBaseCollateral != 0) {
-      FP96.FixedPoint memory baseDebtCoeffMul = FP96.powTaylor(
-        interestRate.mul(FP96.FixedPoint({inner: systemLeverage.shortX96})).div(secondsInYear).add(FP96.one()),
-        secondsPassed
-      );
+      uint256 _discountedBaseDebt = discountedBaseDebt;
+      FP96.FixedPoint memory baseDebtCoeffPrev = baseDebtCoeff;
+      FP96.FixedPoint memory onePlusIR = interestRate
+        .mul(FP96.FixedPoint({inner: systemLeverage.shortX96}))
+        .div(secondsInYear)
+        .add(FP96.one());
+
+      // AR(dt) =  (1+ ir)^dt
+      FP96.FixedPoint memory accruedRateDt = FP96.powTaylor(onePlusIR, secondsPassed);
 
-      FP96.FixedPoint memory baseDebtCoeffOld = baseDebtCoeff;
-      baseDebtCoeff = baseDebtCoeffOld.mul(baseDebtCoeffMul);
       baseCollateralCoeff = baseCollateralCoeff.add(
-        FP96.fromRatio(baseDebtCoeff.sub(baseDebtCoeffOld).mul(discountedBaseDebt), discountedBaseCollateral)
+        FP96.fromRatio(
+          accruedRateDt.sub(FP96.one()).mul(baseDebtCoeffPrev).mul(_discountedBaseDebt),
+          discountedBaseCollateral
+        )
+      );
+
+      baseDebtCoeff = baseDebtCoeffPrev.mul(FP96.powTaylor(onePlusIR.mul(onePlusFee), secondsPassed));
+      discountedBaseFee = baseCollateralCoeff.recipMul(
+        accruedRateDt.mul(feeDt.sub(FP96.one())).mul(baseDebtCoeffPrev).mul(_discountedBaseDebt)
       );
     }
 
     if (discountedQuoteCollateral != 0) {
-      FP96.FixedPoint memory quoteDebtCoeffMul = FP96.powTaylor(
-        interestRate.mul(FP96.FixedPoint({inner: systemLeverage.longX96})).div(secondsInYear).add(FP96.one()),
-        secondsPassed
-      );
+      uint256 _discountedQuoteDebt = discountedQuoteDebt;
+      FP96.FixedPoint memory quoteDebtCoeffPrev = quoteDebtCoeff;
+      FP96.FixedPoint memory onePlusIR = interestRate
+        .mul(FP96.FixedPoint({inner: systemLeverage.longX96}))
+        .div(secondsInYear)
+        .add(FP96.one());
+
+      // AR(dt) =  (1+ ir)^dt
+      FP96.FixedPoint memory accruedRateDt = FP96.powTaylor(onePlusIR, secondsPassed);
 
-      FP96.FixedPoint memory quoteDebtCoeffOld = quoteDebtCoeff;
-      quoteDebtCoeff = quoteDebtCoeffOld.mul(quoteDebtCoeffMul);
       quoteCollateralCoeff = quoteCollateralCoeff.add(
-        FP96.fromRatio(quoteDebtCoeff.sub(quoteDebtCoeffOld).mul(discountedQuoteDebt), discountedQuoteCollateral)
+        FP96.fromRatio(
+          accruedRateDt.sub(FP96.one()).mul(quoteDebtCoeffPrev).mul(_discountedQuoteDebt),
+          discountedQuoteCollateral
+        )
+      );
+
+      quoteDebtCoeff = quoteDebtCoeffPrev.mul(FP96.powTaylor(onePlusIR.mul(onePlusFee), secondsPassed));
```

Comment:
I think it might be better to update debtCoeff as `quoteDebtCoeff = quoteDebtCoeffPrev.mul(accruedRateDt).mul(feeDt)`